File: mobile_app/lib/features/admin/payments/payment.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'payment.freezed.dart';
part 'payment.g.dart';

@freezed
class Payment with _$Payment {
  const factory Payment({
    required String id,
    @JsonKey(name: 'member_id') required String memberId,
    @JsonKey(name: 'subscription_id') required String subscriptionId,
    required double amount,
    required String method,
    @JsonKey(name: 'paid_at') required DateTime paidAt,
    @JsonKey(name: 'created_at') required DateTime createdAt,
  }) = _Payment;

  factory Payment.fromJson(Map<String, dynamic> json) => _$PaymentFromJson(json);
}

@freezed
class PaymentSummary with _$PaymentSummary {
  const factory PaymentSummary({
    @JsonKey(name: 'total_amount') required double totalAmount,
    @JsonKey(name: 'payment_count') required int paymentCount,
    required String month,
  }) = _PaymentSummary;

  factory PaymentSummary.fromJson(Map<String, dynamic> json) =>
      _$PaymentSummaryFromJson(json);
}

File: internal/models/payment.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Payment records a single payment transaction.
type Payment struct {
	ID                uuid.UUID `json:"id"`
	MemberID          uuid.UUID `json:"member_id"`
	SubscriptionID    uuid.UUID `json:"subscription_id"`
	Amount            float64   `json:"amount"`
	Method            string    `json:"method"` // Cash | bKash | Nagad | Card
	PaidAt            time.Time `json:"paid_at"`
	RecordedByAdminID uuid.UUID `json:"recorded_by_admin_id"`
	CreatedAt         time.Time `json:"created_at"`
}

// PaymentSummary is the aggregated result for a given month.
type PaymentSummary struct {
	TotalAmount  float64 `json:"total_amount"`
	PaymentCount int64   `json:"payment_count"`
	Month        string  `json:"month"` // YYYY-MM
}

// PaymentFilter holds optional date-range params for listing payments.
type PaymentFilter struct {
	From *time.Time // nil = no lower bound
	To   *time.Time // nil = no upper bound
}
